internal/shared/resourceclaims.go

```diff
@@ -228,3 +230,137 @@ func GetDRADeviceCELExpressions(device appsv1alpha1.DRADeviceSpec) ([]string, er
+		case resource.ResourceClaimTemplateName != nil:
+			gpuCount, err = GetGPUDeviceCountForClaimTemplate(ctx, client, *resource.ResourceClaimTemplateName, namespace)
+		case resource.ClaimCreationSpec != nil:
+			gpuCount, err = GetGPUDeviceCountForClaimCreationSpec(ctx, client, *resource.ClaimCreationSpec, namespace)
+		}
+
+		if err != nil {
+			return 0, fmt.Errorf("failed to get GPU count for DRA resource at index %d: %w", i, err)
+		}
+
+		totalGPUCount += gpuCount
+	}
+
+	return totalGPUCount, nil
+}
+
+// GetGPUDeviceCountForClaim calculates GPU count for a specific resource claim.
+func GetGPUDeviceCountForClaim(ctx context.Context, client client.Client, resourceClaimName string, namespace string) (int, error) {
+	if resourceClaimName == "" {
+		return 0, fmt.Errorf("resourceClaimName cannot be empty")
+	}
+	if namespace == "" {
+		return 0, fmt.Errorf("namespace cannot be empty")
+	}
+
+	claim, err := k8sutil.GetResourceClaim(ctx, client, resourceClaimName, namespace)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get resource claim %s in namespace %s: %w", resourceClaimName, namespace, err)
+	}
+
+	return getGPUCountFromDeviceRequests(ctx, client, claim.Spec.Devices.Requests)
+}
+
+// GetGPUDeviceCountForClaimTemplate calculates GPU count for a resource claim template.
+func GetGPUDeviceCountForClaimTemplate(ctx context.Context, client client.Client, resourceClaimTemplateName string, namespace string) (int, error) {
+	if resourceClaimTemplateName == "" {
+		return 0, fmt.Errorf("resourceClaimTemplateName cannot be empty")
+	}
+	if namespace == "" {
+		return 0, fmt.Errorf("namespace cannot be empty")
+	}
+
+	claimTemplate, err := k8sutil.GetResourceClaimTemplate(ctx, client, resourceClaimTemplateName, namespace)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get resource claim template %s in namespace %s: %w", resourceClaimTemplateName, namespace, err)
+	}
+
+	return getGPUCountFromDeviceRequests(ctx, client, claimTemplate.Spec.Spec.Devices.Requests)
+}
+
+// GetGPUDeviceCountForClaimCreationSpec calculates GPU count for a claim creation specification.
+func GetGPUDeviceCountForClaimCreationSpec(ctx context.Context, client client.Client, claimCreationSpec appsv1alpha1.DRAClaimCreationSpec, namespace string) (int, error) {
+	if namespace == "" {
+		return 0, fmt.Errorf("namespace cannot be empty")
+	}
+
+	return getGPUCountFromDeviceSpecs(ctx, client, claimCreationSpec.Devices)
+}
+
+// getGPUCountFromDeviceRequests is a helper function to extract GPU count from device requests.
+func getGPUCountFromDeviceRequests(ctx context.Context, client client.Client, requests []resourcev1beta2.DeviceRequest) (int, error) {
+
+	for _, req := range requests {
+		if req.Exactly.DeviceClassName == "" {
+			continue
+		}
+
+		isGPU, err := isNVIDIAGPU(ctx, client, req.Exactly.DeviceClassName)
+		if err != nil {
+			// This allows partial success scenarios
```

Comment:
we would need to fail in all cases here as GPU count might be incorrect, if there is an error getting the device-class for e.g.